access/ms-teams/plugindata.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/teleport-plugins/lib/plugindata"
+)
+
+// PluginData is a data associated with access request that we store in Teleport using UpdatePluginData API.
+type PluginData struct {
+	plugindata.AccessRequestData
+	TeamsData []TeamsMessage
+}
+
+// TeamsMessage represents sent message information
+type TeamsMessage struct {
+	ID          string
+	Timestamp   string
+	RecipientID string
+}
+
+// DecodePluginData deserializes a string map to PluginData struct.
+func DecodePluginData(dataMap map[string]string) interface{} {
+	data := PluginData{}
+
+	data.AccessRequestData = plugindata.DecodeAccessRequestData(dataMap)
+
+	if channelID, timestamp := dataMap["channel_id"], dataMap["timestamp"]; channelID != "" && timestamp != "" {
+		data.TeamsData = append(data.TeamsData, TeamsMessage{ID: channelID, Timestamp: timestamp})
+	}
+	if str := dataMap["messages"]; str != "" {
+		for _, encodedMsg := range strings.Split(str, ",") {
+			parts := strings.Split(encodedMsg, "/")
```

Comment:
Can we assume that ID, Timestamp and RecipientID won't have neither a `,` or a `/`?

Reply:
Yes. ID and RecipientID are both UUIDs, Timestamp is RFC822 if I am not mistaken. But you're right, this possibly needs changing. I've just used the same format we used in other plugins.